### PR TITLE
Clean up the docs

### PR DESCRIPTION
**Pull request by WilliamJamieson**
Clean up the docs for asdf-astropy.

- Rename RTD file because current name was deprecated.
- Make RTD tests more strict (fix resulting errors).

### CODE REVIEW

**Comment by WilliamJamieson** [comment]
This PR needs to wait until the next ASDF release in order to fix the documentation links. So it will stay a draft, so I can revert the temporary change of pointing at the asdf dev branch before this PR gets merged.

**Review by perrygreenfield — APPROVED** [review]
LGTM

**Pull request merged**
